library/base.go

```diff
@@ -0,0 +1,227 @@
+package library
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+type runningProcess struct {
+	cmd     *exec.Cmd
+	sender  Sender
+	output  *SafeBuffer
+	tracker *StatusTracker
+	done    <-chan bool
+	pid     int
+	tempdir string
```

Comment:
What do you need this tempdir for?

Reply:
I thought I would always launch the process in a temp dir and keep it out of the server dir or from other processes.

Reply:
In that case, I would just set `Cmd.Dir = "some/dir"` instead of changing the working directory of the process running the library.